### PR TITLE
Refactor get collection middleware for contacts

### PR DESCRIPTION
**Pull request by yeahfro**
Description of change:
- start at `/src/apps/contacts/middleware/collection.js`
- concept of modules
-- search
-- permissions
-- API
- moving away from horizontal splits such as `middleware`, `filters.js`, `builders.js`

### CODE REVIEW

File: src/modules/api/transformers/api-response-to-collection.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const { buildPagination } = require('../../../lib/pagination')
+
+/**
+ * @param {object} [options] {object}
+ * @param {string} [options.searchTerm] - search term used for highlighting words in collection macro
+ * @param {object} [options.query] - URL query object used in pagination
+ * @param {...function} [itemTransformers] - an array of transformer functions to apply for each item in the list
+ * @returns {object, undefined}
+ */
+function transformApiResponseToCollection (options = {}, ...itemTransformers) {
```
Comment (yozzo) [inline src/modules/api/transformers/api-response-to-collection.js]: Now that we're at the point of refactoring stuff, I find the three callbacks here a bit hard to follow and I think having each function declared individually outside the scope of `transformApiResponseToCollection`  and  only executing them inside would tick my box of clean refactoring. Also passing `items` as an argument to the arrow function, rather than accessing it from the parent function would make it more clear, I'm not a big fan of scope leaks

File: src/modules/search/constants.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+// TODO this does not feel like the right place for this as the search module should not know about apps
```
Comment (yozzo) [inline src/modules/search/constants.js]: it does what it says on the label, `constants`, I'm happy with that.

File: src/modules/search/services.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+const { assign } = require('lodash')
+
+const authorisedRequest = require('../../lib/authorised-request')
+const config = require('../../../config')
+
+function search ({ token, searchTerm = '', searchEntity, requestBody, isAggregation = true, limit = 10, page = 1 }) {
```
Comment (yozzo) [inline src/modules/search/services.js]: aligning arguments in a column would make it more readable.

Comment (ztolley) [reply src/modules/search/services.js]: I started using prettier elsewhere, and it solves discussions like this

File: src/modules/api/transformers/api-response-to-collection.js (hunk 1 above)
Comment (yeahfro) [reply src/modules/api/transformers/api-response-to-collection.js]: Most of this I would agree with but it's outside the scope of this change

File: src/modules/search/constants.js (hunk 2 above)
Comment (yeahfro) [reply src/modules/search/constants.js]: I am not comfortable with this common library having knowledge of the dependent applications

File: src/apps/contacts/middleware/collection.js [hunk 4]
```diff
@@ -1,27 +1,7 @@
...
-    next()
-  } catch (error) {
-    next(error)
-  }
-}
+const { getCollection } = require('../../../middleware/collection')
 
 module.exports = {
-  getContactsCollection,
+  getContactsCollection: getCollection('contact', transformContactToListItem),
```
Comment (yeahfro) [inline src/apps/contacts/middleware/collection.js]: Is everyone happy with this usage? Should we just sack of `collection.js` altogether and do this in `router.js`?

Comment (yozzo) [reply src/apps/contacts/middleware/collection.js]: I'd go for the `router` option

File: src/modules/search/services.js (hunk 3 above)
Comment (ztolley) [reply src/modules/search/services.js]: Going forward I'd advise people to change the lint config and adopt prettier & Airbnb. A config for this can be found in https://github.com/ztolley/blubber/blob/master/package.json

It will be a mahoosive change though as the one thing prettier won't support is `function doStuff (param)` where there is a space between the function name and the bracket, and the author refuses to allow people to do that.

File: src/modules/api/transformers/api-response-to-collection.js (hunk 1 above)
Comment (ztolley) [reply src/modules/api/transformers/api-response-to-collection.js]: I tend to refactor as I touch code, though I know this can cause inconsistent code :/
